### PR TITLE
RM-7468: ProcessUtils.GracefulProcessShutdown does not handle Win32Exception when killing a terminating process

### PR DESCRIPTION
**Pull request by iam-mholle**
https://re-motion.atlassian.net/browse/RM-7468

### CODE REVIEW

File: Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs [hunk 1]
```diff
@@ -143,7 +143,16 @@ public static void GracefulProcessShutdown ([NotNull] Process process, TimeSpan
...
+        process.Kill();
+      }
+      catch (Win32Exception)
+      {
+        // Thrown if the .Kill() method is called while the process is currently terminating.
+        // https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill#remarks
+        // This exception can be swallowed safely, as the process that should be killed is already terminating.
+      }
 
       if (process.WaitForExit (timeoutInMilliseconds))
```
Comment (MichaelKetting) [inline Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs:157]: Hmm...I think if the process is already dead at this point, we might get another exception (see docs). this one should be easy to test with a debugger to delay the call to WaitForExit until the process is gone. don't know if the currently being-killed process will also throw, same as in line 148, though, or if it's just a missing process that will throw. =-> check docs, do trials, make safe.

Comment (iam-mholle) [reply Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs:157]: That's an issue we must have met before the removal of this line: https://github.com/re-motion/Trunk-SVN2Git/pull/113/files#diff-3569e8db6d4a2bdf25ec2682ebd29c24L127

Pretty sure this is a non-issue.

Comment (iam-mholle) [reply Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs:157]: Yep, tested while asserting that the process has exited.

File: Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs [hunk 2]
```diff
@@ -143,7 +143,16 @@ public static void GracefulProcessShutdown ([NotNull] Process process, TimeSpan
       }
 
       // Force closing the process
-      process.Kill();
+      try
+      {
+        process.Kill();
+      }
+      catch (Win32Exception)
```
Comment (MichaelKetting) [inline Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs:150]: One more:
https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.process.kill?view=netframework-4.8

Kill is said to throw an InvalidOpEx if the process has alreayd existed. Please verify.

Comment (MichaelKetting) [reply Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs:150]: Oh, and we should probably doc all the non-issue exceptions regardless if they are mentioned in the docs but do not apply for non-obvious reasons.

Comment (iam-mholle) [reply Remotion/Web/Development.WebTesting/Utilities/ProcessUtils.cs:150]: That's true, I'll add another `catch` block.

**Comment by iam-mholle** [comment]
Added to master with 3e648b9240a3a00c6d87f25194a2be59ae6820bc.

**Pull request closed without merging**